### PR TITLE
Many minor fixes to improve performance and reliability for benchmark

### PR DESCRIPTION
**Pull request by dwightguth**
**please review**

### CODE REVIEW

**Comment by dwightguth** [comment]
Jenkins: test this please

**Comment by dwightguth** [comment]
Jenkins: test this please

**Comment by dwightguth** [comment]
@chathhorn please review

File: semantics/language/common/debug.k [hunk 1]
```diff
@@ -1,5 +1,4 @@
 module DEBUG-SYNTAX
      imports BASIC-K
-     syntax ErrorExit ::= "HALT" | HALT(K)
```
Comment (chathhorn) [inline semantics/language/common/debug.k]: Could we go ahead and remove this file?

**Comment by chathhorn** [comment]
Looks good!

**Pull request merged**